### PR TITLE
7-POST-/api/articles/article_id/comments complete

### PR DESCRIPTION
**Pull request by greg19850**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -212,5 +212,45 @@ describe('app', () => {
...
+    });
+    it('400: GET responds with error, when body is empty', () => {
+      return request(app)
+        .post('/api/articles/1/comments')
+        .send({})
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe('Comment body empty!');
+        });
+    });
```
Comment (Slyose) [inline __tests__/app.test.js:265]: Great testing so far, but we also need tests for:
A status 201 where to request body has unnecessary properties that are ignored 
e.g { author: 'icellusedkars', body: 'All good in the hood', extra: "All goodie in the hoodie" },

A 404 where the id is a number but it's not found e.g id 9999999,

And finally a 404 where the author is not found e.g author: 'Harry Potter'

**Review by Slyose — COMMENTED** [review]
Great work so far, just short a few tests!

**Comment by greg19850** [comment]
I replaced previous Error 400 - author field empty, with 404 - author not found, as PSQL was giving same error code, for empty string and for invalid username.

**Comment by greg19850** [comment]
Just realised one test is incorrect (forgot to replace get method with post). will correct and re-submit pr

File: controllers/errorHandlingControllers.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
 exports.handlePSQL400Errors = (err, req, res, next) => {
   if (err.code === '22P02') {
     res.status(400).send({ msg: 'Invalid Path Request' });
+  } else if (err.code === '23502') {
+    res.status(400).send({ msg: 'Comment body empty!' });
+  } else if (err.code === '23503') {
```
Comment (Slyose) [inline controllers/errorHandlingControllers.js:6]: This code is a generic error that says the key you're looking for doesn't exist - "Author not found" will only work for one endpint.

So you should change the error msg to be more generalized - eg "Invalid Input"

**Pull request merged**